lib/src/puppetdb-cli.cc

```diff
@@ -107,7 +121,85 @@ PuppetDBConn::parseServerUrls(const json::JsonContainer& config) {
     }
 }
 
-size_t write_data(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+class JsonQueue {
```

Comment:
Already discussed this with @rbrw, we'll have to deal with this eventually but removing the fine-grained locking with require us to handle our own buffering in the rapidjson wrapper, similar to what they do for FileReadStream's. This is a non-trivial amount of work and should be done in a different PR.